lte/gateway/c/core/oai/test/ngap/test_ngap_state_converter.cpp

```diff
@@ -246,4 +250,423 @@ TEST_F(NGAPStateConverterTest, NgapStateConversionNgapImsimap) {
+  NGAPClientServicer::getInstance().map_ngap_uestate_proto_str.clear();
+  NGAPClientServicer::getInstance().map_imsi_table_proto_str.clear();
+
+  EXPECT_EQ(
+      hashtable_uint64_ts_get(
+          imsi_map->amf_ue_id_imsi_htbl,
+          (const hash_key_t) ue_ref->amf_ue_ngap_id, &imsi64),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          &state->gnbs, (const hash_key_t) gNB_ref->sctp_assoc_id,
+          reinterpret_cast<void**>(gNB_ref)),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          &state->amfid2associd, (const hash_key_t) ue_ref->amf_ue_ngap_id,
+          reinterpret_cast<void**>(&gNB_ref->sctp_assoc_id)),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_uint64_ts_get(
+          &gNB_ref->ue_id_coll, (const hash_key_t) ue_ref->amf_ue_ngap_id,
+          &comp_ngap_id),
+      HASH_TABLE_OK);
+
+  if (!is_task_state_same) {
+    put_ngap_state();
+  }
+  if (!is_ue_state_same) {
+    put_ngap_imsi_map();
+    put_ngap_ue_state(imsi64);
+  }
+
+  EXPECT_EQ(
+      NGAPClientServicer::getInstance().map_ngap_state_proto_str.isEmpty(),
+      false);
+  EXPECT_EQ(
+      NGAPClientServicer::getInstance().map_ngap_uestate_proto_str.isEmpty(),
+      true);
+  EXPECT_EQ(
+      NGAPClientServicer::getInstance().map_imsi_table_proto_str.isEmpty(),
+      true);
+
+  EXPECT_EQ(
+      NGAPClientServicer::getInstance().map_ngap_state_proto_str.size(), 1);
+  EXPECT_EQ(
+      NGAPClientServicer::getInstance().map_ngap_uestate_proto_str.size(), 0);
+  EXPECT_EQ(
+      NGAPClientServicer::getInstance().map_imsi_table_proto_str.size(), 0);
+
+  NgapStateConverterTest::TearDown();
+
+  EXPECT_EQ(state, nullptr);
+  EXPECT_EQ(imsi_map, nullptr);
+
+  NgapStateConverterTest::SetUp();
+
+  gnb_description_t* get_gNB_ref = NULL;
+  // Check if gNB exists
+  get_gNB_ref = ngap_state_get_gnb(state, peerInfo.assoc_id);
+  ASSERT_TRUE(get_gNB_ref != NULL);
+
+  EXPECT_NE(state, nullptr);
+  EXPECT_NE(imsi_map, nullptr);
+  EXPECT_EQ(
+      hashtable_ts_is_key_exists(
+          &state->gnbs, (const hash_key_t) gNB_ref->sctp_assoc_id),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_ts_is_key_exists(
+          &state->amfid2associd, (const hash_key_t) ue_ref->amf_ue_ngap_id),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_uint64_ts_is_key_exists(
+          &get_gNB_ref->ue_id_coll, (const hash_key_t) ue_ref->amf_ue_ngap_id),
+      HASH_TABLE_OK);
+  EXPECT_EQ(
+      hashtable_uint64_ts_is_key_exists(
+          imsi_map->amf_ue_id_imsi_htbl,
+          (const hash_key_t) ue_ref->amf_ue_ngap_id),
+      HASH_TABLE_KEY_NOT_EXISTS);
+
+  ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_Ngap_NGAP_PDU, &decoded_pdu);
+  itti_free_msg_content(message_p);
+  free(message_p);
+  bdestroy_wrapper(&ran_cp_ipaddr);
+  bdestroy(buffer);
+  bdestroy(ngap_initial_ue_msg);
+}
+
+// 2. Stateless feature Unit Test for NGAP
+TEST_F(NgapStateConverterTest, test_ngap_stateless) {
```

Comment:
Please use camelcase for all test names

```suggestion
TEST_F(NgapStateConverterTest, TestNgapServiceRestart) {
```

Reply:
Addressed review comment